jobs/rocoto/prep.sh

```diff
@@ -1,129 +1,141 @@
    status=$?
-   [[ $status -ne 0 ]] && exit $status
+   [[ ${status} -ne 0 ]] && exit ${status}
 
 #  Ensure previous cycle gdas dumps are available (used by cycle & downstream)
-   GDATE=$($NDATE -$assim_freq $CDATE)
-   gPDY=$(echo $GDATE | cut -c1-8)
-   gcyc=$(echo $GDATE | cut -c9-10)
+   GDATE=$(${NDATE} -${assim_freq} ${CDATE})
+   gPDY=$(echo ${GDATE} | cut -c1-8)
+   gcyc=$(echo ${GDATE} | cut -c9-10)
    GDUMP=gdas
-   gCOMOUT="$ROTDIR/$GDUMP.$gPDY/$gcyc/$COMPONENT"
-   if [ ! -s $gCOMOUT/$GDUMP.t${gcyc}z.updated.status.tm00.bufr_d ]; then
-     $HOMEgfs/ush/getdump.sh $GDATE $GDUMP $DMPDIR/${GDUMP}${DUMP_SUFFIX}.${gPDY}/${gcyc}/${COMPONENT} $gCOMOUT
+   gCOMOBS="${ROTDIR}/${GDUMP}.${gPDY}/${gcyc}/obs"
+   if [[ ! -s ${gCOMOBS}/${GDUMP}.t${gcyc}z.updated.status.tm00.bufr_d ]]; then
+     ${HOMEgfs}/ush/getdump.sh "${GDATE}" "${GDUMP}" "${DMPDIR}/${GDUMP}${DUMP_SUFFIX}.${gPDY}/${gcyc}/${COMPONENT}" "${gCOMOBS}"
      status=$?
-     [[ $status -ne 0 ]] && exit $status
+     [[ ${status} -ne 0 ]] && exit ${status}
    fi
-
+   # exception handling to ensure no dead link
+   [[ $(find ${COMIN_OBS} -xtype l | wc -l) -ge 1 ]] && exit 9
+   [[ $(find ${gCOMOBS} -xtype l | wc -l) -ge 1 ]] && exit 9
 fi
 
+
 ###############################################################
 
 ###############################################################
 # For running real-time parallels, execute tropcy_qc and
 # copy files from operational syndata directory to a local directory.
 # Otherwise, copy existing tcvital data from globaldump.
 
-if [ $PROCESS_TROPCY = "YES" ]; then
+if [[ ${PROCESS_TROPCY} = "YES" ]]; then
 
     export COMINsyn=${COMINsyn:-$(compath.py gfs/prod/syndat)}
-    if [ $RUN_ENVIR != "nco" ]; then
+    if [[ ${RUN_ENVIR} != "nco" ]]; then
         export ARCHSYND=${ROTDIR}/syndat
-        if [ ! -d ${ARCHSYND} ]; then mkdir -p $ARCHSYND; fi
-        if [ ! -s $ARCHSYND/syndat_akavit ]; then
+        if [[ ! -d ${ARCHSYND} ]]; then mkdir -p ${ARCHSYND}; fi
+        if [[ ! -s ${ARCHSYND}/syndat_akavit ]]; then
             for file in syndat_akavit syndat_dateck syndat_stmcat.scr syndat_stmcat syndat_sthisto syndat_sthista ; do
-                cp $COMINsyn/$file $ARCHSYND/.
+                cp ${COMINsyn}/${file} ${ARCHSYND}/.
             done
         fi
     fi
 
-    [[ $ROTDIR_DUMP = "YES" ]] && rm $COMOUT${CDUMP}.t${cyc}z.syndata.tcvitals.tm00
+    [[ ${ROTDIR_DUMP} = "YES" ]] && rm ${COMOUT}${CDUMP}.t${cyc}z.syndata.tcvitals.tm00
 
-    $HOMEgfs/jobs/JGLOBAL_ATMOS_TROPCY_QC_RELOC
+    ${HOMEgfs}/jobs/JGLOBAL_ATMOS_TROPCY_QC_RELOC
     status=$?
-    [[ $status -ne 0 ]] && exit $status
+    [[ ${status} -ne 0 ]] && exit ${status}
 
 else
-    [[ $ROTDIR_DUMP = "NO" ]] && cp $DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT}/${CDUMP}.t${cyc}z.syndata.tcvitals.tm00 $COMOUT/
+    [[ ${ROTDIR_DUMP} = "NO" ]] && cp ${DMPDIR}/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT}/${CDUMP}.t${cyc}z.syndata.tcvitals.tm00 ${COMOUT}/
```

Comment:
```suggestion
    [[ ${ROTDIR_DUMP} = "NO" ]] && cp ${DMPDIR}/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/atmos/${CDUMP}.t${cyc}z.syndata.tcvitals.tm00 ${COMOUT}/
```
Isn't this *always* in the `atmos` component.  If so, force it.